client/my-sites/upgrades/domain-management/transfer/icann-verification.jsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import SectionHeader from 'components/section-header';
+import { resendIcannVerification } from 'lib/upgrades/actions/domain-management';
+import Button from 'components/button';
+import notices from 'notices';
+
+const IcannVerification = React.createClass( {
+	getInitialState() {
+		return {
+			submitting: false
+		};
+	},
+
+	handleClick() {
+		this.setState( { submitting: true } );
+
+		resendIcannVerification( this.props.selectedDomainName, ( error ) => {
+			if ( error ) {
+				notices.error( error.message );
+			} else {
+				notices.success( this.translate(
+					'We successfully resent the ICANN verification email to your ' +
+					'email address on file.'
+				) );
+			}
+
+			this.setState( { submitting: false } );
+		} );
+	},
+
+	render() {
+		return (
+			<div>
+				<SectionHeader label={ this.translate( 'Transfer Domain' ) }>
+					<Button
+						onClick={ this.handleClick }
+						disabled={ this.state.submitting }
+						compact
+						primary={ true }>{ this.translate( 'Resend Verification Email' ) }</Button>
+				</SectionHeader>
+
+				<Card className="transfer-card">
+					<p>
+						{ this.translate(
+							'You must verify your email address before you can transfer this domain. ' +
+							'{{learnMoreLink}}Learn more.{{/learnMoreLink}}',
+							{
+								components: {
+									learnMoreLink: <a
```

Comment:
Yup, I think de-duplication and ease of refactoring are the key benefits here. I'd treat these URLs as any other hardcoded value and, well, avoid doing it. Especially since we have these support URLs sprinkled across our code. Of course, this is not an issue just with this PR, it just happens that a related issue was raised recently :)